homeassistant/components/azure_data_explorer/config_flow.py

```diff
@@ -0,0 +1,125 @@
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ADX_CLUSTER_INGEST_URI): str,
+        vol.Required(CONF_ADX_DATABASE_NAME): str,
+        vol.Required(CONF_ADX_TABLE_NAME): str,
+        vol.Required(CONF_APP_REG_ID): str,
+        vol.Required(CONF_APP_REG_SECRET): str,
+        vol.Required(CONF_AUTHORITY_ID): str,
+        vol.Optional(CONF_USE_FREE, default=False): bool,
+    }
+)
+
+
+class ADXConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Azure Data Explorer."""
+
+    VERSION = 1
+
+    async def validate_input(self, data: dict[str, Any]) -> dict[str, Any] | None:
+        """Validate the user input allows us to connect.
+
+        Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+        """
+        client = AzureDataExplorerClient(data)
+
+        try:
+            await self.hass.async_add_executor_job(client.test_connection)
+
+        except KustoAuthenticationError as exp:
+            _LOGGER.error(exp)
+            return {"base": "invalid_auth"}
+
+        except KustoServiceError as exp:
+            _LOGGER.error(exp)
+            return {"base": "cannot_connect"}
+
+        except Exception as exp:  # pylint: disable=broad-except
+            _LOGGER.error(exp)
+            return {"base": "unknown"}
+
+        return None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> ADXOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return ADXOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors: dict = {}
+        if user_input:
+            errors = await self.validate_input(user_input)  # type: ignore[assignment]
+            if not errors:
+                return self.async_create_entry(
+                    data=user_input,
+                    title=user_input[CONF_ADX_CLUSTER_INGEST_URI].replace(
+                        "https://", ""
+                    ),
+                    options=DEFAULT_OPTIONS,
+                )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=STEP_USER_DATA_SCHEMA,
+            errors=errors,
+            last_step=True,
+        )
+
+
+class ADXOptionsFlowHandler(config_entries.OptionsFlowWithConfigEntry):
+    """Handle azure adx options."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the ADX options."""
+        if user_input is not None:
+            return self.async_create_entry(data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SEND_INTERVAL,
```

Comment:
The `CONF_SEND_INTERVAL` was discussed internally, and we concluded that you should use a sane default value and remove the option to configure. Same as the update_interval should not be specified via an option flow also, the send interval should not be configurable.
Please remove it

Reply:
I have removed the update parameter, but instead i have added the secrct as an parameter. the secret has a max life up 2 years, so it needs to be updated.